service_restaurant/src/main/kotlin/org/thechance/service_restaurant/datasource/RestaurantDataSource.kt

```diff
@@ -0,0 +1,21 @@
+package org.thechance.service_restaurant.datasource
+
+import org.koin.core.annotation.Single
+import org.thechance.service_restaurant.entity.Restaurant
+
+@Single
+class RestaurantDataSource(private val dataBase: DataBase) {
+
+    private val restaurantCollection by lazy {
```

Comment:
```suggestion
    private val collection by lazy {
```
because we are in RestaurantDataSource

Reply:
@mosayed01 
The collection is similar to a Table, and as the system progresses, there will be multiple collections with the same name😅so I suggest that the name 'restaurantCollection' could simply be 'restaurant or keeping the same name